media/js/base/mozilla-client.js

```diff
@@ -320,6 +320,71 @@ if (typeof Mozilla === 'undefined') {
         }));
     };
 
+    /**
+     * Use the async mozUITour API of Firefox to retrieve the user's Sync/FxA info. This API
+     * is available on Firefox 35 and later. See
+     * http://bedrock.readthedocs.org/en/latest/uitour.html for details.
+     *
+     * @param  {Function} callback - callback function to be executed with the Sync details
+     * @return {None}
+     */
+    Client.getSyncDetails = function (callback) {
+        // Fire the callback function immediately if cache exists
+        if (Client.SyncDetails) {
+            callback(Client.SyncDetails);
+
+            return;
+        }
+
+        var callbackID = Math.random().toString(36).replace(/[^a-z]+/g, '');
+
+        var listener = function (event) {
+            if (!event.detail || !event.detail.data || event.detail.callbackID !== callbackID) {
+                return;
+            }
+
+            window.clearTimeout(timer);
+            onRetrieved(event.detail.data.setup, event.detail.data.desktopDevices, event.detail.data.mobileDevices, event.detail.data.totalDevices);
+        };
+
+        var onRetrieved = function (setup, desktopDevices, mobileDevices, totalDevices) {
+            document.removeEventListener('mozUITourResponse', listener, false);
+
+            var details = Client.SyncDetails = {
+                'setup': setup,
+                'desktopDevices': desktopDevices,
```

Comment:
There's always a chance that these values could be undefined, since they are only supported in Firefox 50 and above. Should we fallback to `0` if not supported?

Reply:
Definitely. Will do.